### PR TITLE
First API routes

### PR DESCRIPTION
**Pull request by 3wnbr1**
This one adds new routes to the application. Provivided API has yet limited functionnalities but allows search by vendor/product, description and allow to fetch vulnerabilities as JSON, detailled or not.

API routes and roles are detailled bellow.

| URL route                                                                                             | Role                                                                      | HTTP methods       |
|-------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------|--------------------|
| `/api/v1/<cve_id>`                                                                                    | View vuln as JSON                                                         | GET, HEAD, OPTIONS |
| `/api/v1/details/<cve_id>`                                                                            | View detailled vuln as JSON                                               | GET, HEAD, OPTIONS |
| `/api/v1/product/<vendor_id>/<product_id>`                                                            | Search CVE ids by vendor and product                                      | GET, HEAD, OPTIONS |
| `/api/v1/search/description:<description>`                                                            | Search CVE ids by description                                             | GET, HEAD, OPTIONS |
| `/api/v1/search/vendor:<name>`                                                                        | Search vendor by name                                                     | GET, HEAD, OPTIONS |
| `/api/v1/search/product:<name>`                                                                       | Search product by name                                                    | GET, HEAD, OPTIONS |
| `/api/v1/search/vendor_or_product:<name>` `/api/v1/search/product_or_vendor:<name>`                   | Search products where vendor or product matches name                      | GET, HEAD, OPTIONS |
| `/api/v1/search/vendor:<vendor>/product:<product>` `/api/v1/search/product:<product>/vendor:<vendor>` | Search products where vendors matches vendor and products matches product | GET, HEAD, OPTIONS |

PR is mostly for review and not yet mature to be merged. Feedbacks are welcome.

### CODE REVIEW

File: app/api/v1/routes.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    cve = db.session.query(Nvd.cve_id).filter(Nvd.id.in_(nvd_ids)).all()
+    return jsonify({"count": count, "cve_ids": [x for x, in cve]})
+
+
+def _cpesToJson(products):
+    """Jsonify Cpes for API routes."""
+    count = len(products)
+    return jsonify({
+        "count":
+        count,
```
Comment (bluec0re) [inline app/api/v1/routes.py]: Could you put this in the same line with the key?

File: data/models/nvd.py [hunk 2]
```diff
@@ -26,35 +26,103 @@
 
 
 class Affect(nvd_template.Affect, NvdBase):
-    pass
+    def toJson(self):
```
Comment (bluec0re) [inline data/models/nvd.py]: I know we're still inconsistent, but we would like to be pep8 compliant at some point 😅. So snakecase for naming would be great 🙂. (Same for the other methods below)

File: app/api/v1/routes.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        return abort(404)
+    nvd_ids = db.session.query(Cpe.nvd_json_id).filter(
+        and_(Cpe.vendor == vendor_id,
+             Cpe.product == product_id)).distinct().all()
+    count = len(nvd_ids)
+    cve = db.session.query(Nvd.cve_id).filter(Nvd.id.in_(nvd_ids)).all()
+    return jsonify({"count": count, "cve_ids": [x for x, in cve]})
+
+
+def _cpesToJson(products):
```
Comment (bluec0re) [inline app/api/v1/routes.py]: Could you make this snake_case as well?

**Comment by evonide** [comment]
This looks great, thank you so much Ewen!
Looking forward to deploy and try it out soon :).

File: app/api/v1/routes.py (hunk 1 above)
Comment (3wnbr1) [reply app/api/v1/routes.py]: That would be great but unfortunatly `yapf` formatter sets it like that all the time. As I was unable to find any options, I simply specified yapf to ignore block like this.

```python
    return jsonify({
        "count": count,
        "products": [{
            'product': x,
            'vendor': y
        } for x, y, in products]
    })  # yapf: disable
```
Certainly not the best way but gets the job done

**Comment by evonide** [comment]
I've fixed some very small things in 4f2f69eb15432635a4cbdd756adee503b34fd923 and have pushed this to prod. Some examples:
- https://www.vulncode-db.com/api/v1/CVE-2013-7470
- https://www.vulncode-db.com/api/v1/details/CVE-2013-7470
- https://www.vulncode-db.com/api/v1/search/vendor:microsoft
- https://www.vulncode-db.com/api/v1/search/product:kernel
- https://www.vulncode-db.com/api/v1/search/vendor:microsoft/product:windows

Again, amazing work Ewen thanks a lot for helping out, this is really appreciated.

**Pull request merged**